modules/eks/outputs.tf

```diff
@@ -0,0 +1,49 @@
+locals {
+  kubeconfig          = <<KUBECONFIG
+apiVersion: v1
+clusters:
+- cluster:
+    server: ${aws_eks_cluster.demo.endpoint}
+    certificate-authority-data: ${aws_eks_cluster.demo.certificate_authority[0].data}
+  name: kubernetes
+contexts:
+- context:
+    cluster: kubernetes
+    user: aws
+  name: aws
+current-context: aws
+kind: Config
+preferences: {}
+users:
+- name: aws
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1alpha1
+      command: heptio-authenticator-aws
+      args:
+        - "token"
+        - "-i"
+        - "${var.cluster-name}"
+KUBECONFIG
+
+  config-map-aws-auth = <<CONFIGMAPAWSAUTH
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: aws-auth
+  namespace: kube-system
+data:
+  mapRoles: |
+    - rolearn: ${aws_iam_role.demo-node.arn}
+      username: system:node:{{EC2PrivateDNSName}}
+      groups:
+        - system:bootstrappers
+        - system:nodes
+CONFIGMAPAWSAUTH
+}
+output "kubeconfig" {
+  value = local.kubeconfig
+}
```

Comment:
you don't need to generate and output your kubeconfig. instead, it's easier you use `aws eks update-kubeconfig` command: https://docs.aws.amazon.com/cli/latest/reference/eks/update-kubeconfig.html

Reply:
By using aws eks update-kubeconfig, you don't need to worry about the certificate-authority-data stuffs?